cocos/2d/components/sprite.ts

```diff
@@ -564,9 +565,14 @@ export class Sprite extends UIRenderer {
     protected _updateBuiltinMaterial (): Material {
         let mat = super._updateBuiltinMaterial();
         if (this.spriteFrame && this.spriteFrame.texture instanceof RenderTexture) {
-            const renderMat = new Material();
-            renderMat.copy(mat, { defines: { SAMPLE_FROM_RT: true } });
-            mat = renderMat;
+            const rtMatName = `rt-${mat.name}`;
```

Comment:
Would the `mat.name` would be empty? If it's empty, `rtMatName` will be `rt-` which will cause issues.

Reply:
If it's possible empty, it's better to check the name, keep the old logic if it's empty.

Reply:
The original code logic does not consider the possibility of mat being null, since built-in resources are generally not deleted.
In the engine's core code, there's no implementation that actively deletes assets from builtinResMgr.
I believe it's unnecessary to account for potential errors caused by engine modifications through unconventional channels.

Reply:
That makes sense.